Raven.Database/FileSystem/Storage/Esent/Schema/Updates/From08To09.cs

```diff
@@ -0,0 +1,189 @@
+                //this is a precaution against partially failed upgrade process
+            }
+
+            SchemaCreator.CreatePagesTable(dbid, newTableName, session);
+
+            var buffer = new byte[StorageConstants.MaxPageSize];
+            var bookMarkBuffer = new byte[SystemParameters.BookmarkMost];
+
+            using (var usage = new Table(session, dbid, "usage", OpenTableGrbit.None))
+            {
+                // let's create index in usages table necessary for migration purposes
+                // we need to seek by page_id in order to update value in 'usages' table after
+                // writing a page to new table where it will get new id
+
+                var indexDef = "+page_id\0\0";
+                try
+                {
+                    Api.JetDeleteIndex(session, usage, "by_page_id");
+                }
+                catch (Exception)
+                {
+                    //if there is no such index - then it is not important
+                    //this is a precaution against partially failed upgrade process
+                }
+
+                Api.JetCreateIndex2(session, usage, new[]
+                {
+                    new JET_INDEXCREATE
+                    {
+                        szIndexName = "by_page_id",
+                        cbKey = indexDef.Length,
+                        cbKeyMost = SystemParameters.KeyMost,
+                        cbVarSegMac = SystemParameters.KeyMost,
+                        szKey = indexDef,
+                        grbit = CreateIndexGrbit.None,
+                        ulDensity = 80
+                    }
+                }, 1);
+
+                // commit index creation
+
+                Api.JetCommitTransaction(session, CommitTransactionGrbit.LazyFlush);
+                Api.JetBeginTransaction2(session, BeginTransactionGrbit.None);
+
+                var usageColumns = Api.GetColumnDictionary(session, usage);
+
+                using (var src = new Table(session, dbid, pagesTableName, OpenTableGrbit.None))
+                using (var dst = new Table(session, dbid, newTableName, OpenTableGrbit.None))
+                { 
+                    Api.MoveBeforeFirst(session, src);
+                    Api.MoveBeforeFirst(session, dst);
+
+                    var srcColumns = Api.GetColumnDictionary(session, src);
+                    var dstColumns = Api.GetColumnDictionary(session, dst);
+
+                    int rows = 0;
+
+                    while (Api.TryMoveNext(session, src))
+                    {
+                        using (var insert = new Update(session, dst, JET_prep.Insert))
+                        {
+                            var oldPageId = Api.RetrieveColumnAsInt32(session, src, srcColumns["id"]).Value;
+
+                            // page_strong_hash
+
+                            using (var srcStream = new BufferedStream(new ColumnStream(session, src, srcColumns["page_strong_hash"])))
```

Comment:
page_strong_hash should be a small buffer, no?

Reply:
It should. Should any code change be made here then?

Reply:
Doesn't matter that much, was just surpirsing